sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -211,4 +215,80 @@ trait ExpressionEvalHelper {
       plan(inputRow)).get(0, expression.dataType)
     assert(checkResult(actual, expected))
   }
+
+  def checkConsistency(dt: DataType, clazz: Class[_]): Unit = {
```

Comment:
What do you think about giving this a more specific name, such as `checkConsistencyBetweenInterpretedAndCodegen`? It would also be good to add Scaladoc to these methods to explain what they're doing, since the use of reflection might be non-obvious.